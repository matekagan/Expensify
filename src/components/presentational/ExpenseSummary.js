import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../../selectors/expenses';
import totalExpenses from '../../selectors/expenses-total';

export const ExpenseSummary = ({ totalAmount, expenseCount }) => (
    <div>
        <h4>
            {
                expenseCount === 0
                    ? 'No expenses to show'
                    : `Viewing 
                ${expenseCount} 
                expense
                ${expenseCount === 1 ? '' : 's'} 
                totalling 
                ${numeral(totalAmount / 100).format('$0,0.00')}`
            }
        </h4>
    </div>
);

const mapStateToProps = state => ({
    ...totalExpenses(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpenseSummary);
