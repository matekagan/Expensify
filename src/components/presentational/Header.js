import React from 'react';
import {
    NavLink
} from 'react-router-dom';

const Header = () => (
    <div>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="navlink-active" exact={true}>Home Page</NavLink>
        <NavLink to="/create" activeClassName="navlink-active">Create Expense</NavLink>
        {/* <NavLink to="/edit" activeClassName="navlink-active">Edit Expense</NavLink> */}
        <NavLink to="/help" activeClassName="navlink-active">Help</NavLink>
    </div>
);

export default Header;
