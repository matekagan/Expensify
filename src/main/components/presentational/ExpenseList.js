import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../../selectors/expenses';

export const ExpenseList = ({ expenses }) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-mobile">
                Expenses
            </div>
            <div className="show-desktop">
                Expense
            </div>
            <div className="show-desktop">
                Amount
            </div>
        </div>
        <div className="list-body">
            {
                expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span> No Expenses to Show</span>
                    </div>
                ) : expenses.map(expense => (
                    <ExpenseListItem key={expense.id} {...expense} />
                ))
            }
        </div>
    </div>
);

const mapStateToProps = state => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);
