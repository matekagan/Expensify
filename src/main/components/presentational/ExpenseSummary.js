import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../../selectors/expenses';
import totalExpenses from '../../selectors/expenses-total';

export const ExpenseSummary = ({ totalAmount, expenseCount }) => {
    const formattedNumber = numeral(totalAmount / 100).format('$0,0.00');
    const pluralDependentText = expenseCount === 1 ? 'expense ' : 'expenses ';
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing
                    <span> {expenseCount} </span>
                    {pluralDependentText}
                    totalling
                    <span> {formattedNumber}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create" activeClassName="navlink-active">Create Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = state => ({
    ...totalExpenses(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpenseSummary);
