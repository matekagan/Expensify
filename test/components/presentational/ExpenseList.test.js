import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../../src/components/presentational/ExpenseList';
import testExpenses from '../../fixtures/expenses';

test('Should render expenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={testExpenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should render expenses list without expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});
