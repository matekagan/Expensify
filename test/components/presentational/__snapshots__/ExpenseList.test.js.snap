// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={200}
    createdAt={2592000000}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={20000}
    createdAt={0}
    description="Gas bill"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={2000}
    createdAt={1296000000}
    description="Water Bill"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`Should render expenses list without expenses 1`] = `
<div>
  <p>
     No Expenses to Show
  </p>
</div>
`;
